app/main.py

```diff
@@ -1,25 +1,72 @@
 from abc import ABC
+from typing import Type
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Type[object], name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: Type[object]) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> Exception | int:
```

Comment:
The return type of the `__set__` method should not be `Exception | int`. This method should not return anything; it should only raise exceptions if the value is invalid.